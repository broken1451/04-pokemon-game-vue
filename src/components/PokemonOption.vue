<template lang="">
    <div>
       <div  class="options-container">
          <!-- <ul :disabled="blocked"> -->
          <ul>
            <!-- <li v-for="(pokemon, index) in pokemonArr" :key="pokemon.id" @click="$emit('nombreDelEventoQuequieroDisparar')"> -->
            <li  v-for="(pokemon, index) in pokemonArr" :key="pokemon.id" @click="$emit('selection', pokemon.id)">
              <span>
              {{capitalizeFirstLetter(pokemon.name) }}
              </span>
            </li>
          </ul>
       </div>
    </div>
</template>
<script>
export default {
  props: {
    pokemonArr: {
      type: Array,
      required: true,
      default: [],
    },
    blocked: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
  data() {
    return {
      isDisabled: this.blocked,
    };
  },
  name: "PokemonOption",
  methods: {
    capitalizeFirstLetter(pokemon = "") {
      // console.log(pokemon);
      pokemon = pokemon.trim();
      pokemon = pokemon.charAt(0).toUpperCase() + pokemon.slice(1);
      return pokemon;
    },
    hola() {
      // this.$emit(event, ${[â€¦args]})
    },
  },
};
</script>
<style scoped>
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>