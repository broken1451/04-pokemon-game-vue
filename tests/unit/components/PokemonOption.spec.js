import { shallowMount } from "@vue/test-utils";
import PokemonOption from "@/components/PokemonOption.vue";
import { mockPokemos } from "../mocks/mockPokemons";

describe("PokemonOption component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOption, {
      props: {
        pokemonArr: mockPokemos,
      },
    });
  });

  it("debe de hacer match con el snapshot ", () => {
    //   console.log(wrapper.html());
    // expect(wrapper.html())..toMatchInlineSnapshot();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("debe de mostrar las 4 opciones correctamente", () => {
    // console.log(wrapper.html());
    const listado = wrapper.findAll("li");
    expect(listado.length).toBe(4);
    expect(listado[0].text()).toBe("Bulbasaur");
    expect(listado[1].text()).toBe("Ivysaur");
    expect(listado[2].text()).toBe("Venusaur");
    expect(listado[3].text()).toBe("Charmander");
  });

  it("debe emitir 'selection' con sus respectivos parametros  al hacer click", () => {
    const [l1, l2, l3, l4] = wrapper.findAll("li");

    l1.trigger("click");
    l2.trigger("click");
    l3.trigger("click");
    l4.trigger("click");
    console.log(wrapper.emitted("selection"));
    expect(wrapper.emitted("selection").length).toBe(4);
    // expect(wrapper.emitted("selection")[0]).toBe([1]);
    expect(wrapper.emitted("selection")[0]).toStrictEqual([1]);
    expect(wrapper.emitted("selection")[1]).toStrictEqual([2]);
    expect(wrapper.emitted("selection")[2]).toStrictEqual([3]);
    expect(wrapper.emitted("selection")[3]).toStrictEqual([4]);
  });
});
