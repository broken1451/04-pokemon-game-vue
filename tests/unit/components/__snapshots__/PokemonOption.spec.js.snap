// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOption component debe de hacer match con el snapshot  1`] = `
<div>
  <div class="options-container">
    <!-- <ul :disabled="blocked"> -->
    <ul>
      <!-- <li v-for="(pokemon, index) in pokemonArr" :key="pokemon.id" @click="$emit('nombreDelEventoQuequieroDisparar')"> -->
      <li><span>Bulbasaur</span></li>
      <li><span>Ivysaur</span></li>
      <li><span>Venusaur</span></li>
      <li><span>Charmander</span></li>
    </ul>
  </div>
</div>
`;
